bcipy/language/main.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Tuple, List, Union
+from enum import Enum
+
+
+class ResponseType(Enum):
+    SYMBOL = 'Symbol'
+    WORD = 'Word'
+
+
+class LanguageModel(ABC):
+
+    response_type: ResponseType
+    symbol_set: List[str]
+
+    @abstractmethod
+    def predict(self, evidence: List[Tuple]) -> List[tuple]:
+        """
+        Using the provided data, compute log likelihoods over the entire symbol set.
+        Args:
+            evidence - [('A', .08), ('B', .1)]
+
+        Response:
+            probability - dependant on response type, a list of words or symbols with probability 
+        """
+        ...
+
+    @abstractmethod
+    def update(self) -> None:
+        """Update the model state"""
+        ...
+
+    @abstractmethod
+    def load(self, path: Path) -> None:
+        """Restore model state from the provided checkpoint"""
+        ...
```

Comment:
We may want an optional method to `reset` any state.

Reply:
Good idea! I'll add it